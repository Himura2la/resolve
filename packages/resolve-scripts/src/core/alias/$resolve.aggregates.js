import resolveFile from '../resolve_file'
import resolveFileOrModule from '../resolve_file_or_module'
import importBabel from '../import_babel'

export default ({ resolveConfig, isClient }) => {
  const imports = []
  const constants = [``]
  const exports = [``, `const aggregates = []`, ``]

  for (let index = 0; index < resolveConfig.aggregates.length; index++) {
    const aggregate = resolveConfig.aggregates[index]

    const name = aggregate.name

    const commands = resolveFile(aggregate.commands)

    const projection = aggregate.projection
      ? resolveFile(aggregate.projection)
      : undefined

    const snapshotAdapter = aggregate.snapshot
      ? resolveFileOrModule(aggregate.snapshot.adapter)
      : undefined

    const snapshotOptions = aggregate.snapshot
      ? aggregate.snapshot.options
      : undefined

    if (!isClient) {
      imports.push(`import commands_${index} from "${commands}"`)
    }

    if (!isClient && aggregate.projection) {
      imports.push(`import projection_${index} from "${projection}"`)
    }

    if (!isClient && aggregate.snapshot) {
      imports.push(`import snapshotAdapter_${index} from "${snapshotAdapter}"`)
    }

    constants.push(`const name_${index} = ${JSON.stringify(name)}`)

    if (isClient) {
      const clientCommands = Object.keys(importBabel(commands))

      constants.push(
        `const commands_${index} = {`,
        clientCommands
          .map(commandName => `  ${commandName}() {}`)
          .join(',\r\n'),
        `}`
      )
    }

    if (!isClient && aggregate.snapshot) {
      constants.push(
        `const snapshotOptions_${index} = ${JSON.stringify(snapshotOptions)}`
      )
    }

    exports.push(`aggregates.push({`)
    exports.push(`  name: name_${index}`)
    exports.push(`, commands: commands_${index}`)
    if (!isClient && aggregate.projection) {
      exports.push(`, projection: ${JSON.stringify(projection)}`)
    }
    if (!isClient && aggregate.snapshot) {
      exports.push(
        `, snapshot: {`,
        `    adapter: snapshotAdapter_${index},`,
        `    options: snapshotOptions_${index}`,
        `  }`
      )
    }
    exports.push(`})`, ``)
  }

  exports.push(`export default aggregates`)

  return {
    code: [...imports, ...constants, ...exports].join('\r\n')
  }
}
