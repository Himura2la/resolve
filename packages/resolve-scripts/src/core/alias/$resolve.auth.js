import { message } from '../constants'

export default ({ resolveConfig, isClient }) => {
  if (isClient) {
    throw new Error(`${message.serverAliasInClientCodeError}$resolve.auth`)
  }

  if (!resolveConfig.auth) {
    throw new Error(`${message.configNotContainSectionError}.auth`)
  }

  const exports = []

  exports.push(
    `import strategies from ${JSON.stringify(resolveConfig.auth.strategies)}`,
    ``,
    `const auth = {`,
    `  strategies`,
    `}`,
    ``,
    `export default auth`
  )

  return {
    code: exports.join('\r\n')
  }
}
