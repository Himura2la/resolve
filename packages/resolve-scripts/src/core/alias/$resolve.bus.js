import { injectEnv } from 'json-env-extract'

import { message } from '../constants'

export default ({ resolveConfig, isClient }) => {
  if (isClient) {
    throw new Error(`${message.serverAliasInClientCodeError}$resolve.bus`)
  }

  if (!resolveConfig.bus) {
    throw new Error(`${message.configNotContainSectionError}.bus`)
  }

  const exports = []

  exports.push(
    `import interopRequireDefault from "@babel/runtime/helpers/interopRequireDefault"`,
    ``,
    `const bus = ${injectEnv(resolveConfig.bus)}`,
    `const adapter = interopRequireDefault(require(bus.adapter)).default`,
    `const options = bus.options`,
    ``,
    `export default { adapter, options }`
  )

  return {
    code: exports.join('\r\n')
  }
}
