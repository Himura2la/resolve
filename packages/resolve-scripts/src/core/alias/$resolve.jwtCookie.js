import { message } from '../constants'

export default ({ resolveConfig, isClient }) => {
  if (isClient) {
    throw new Error(`${message.serverAliasInClientCodeError}$resolve.jwtCookie`)
  }

  if (!resolveConfig.jwtCookie) {
    throw new Error(`${message.configNotContainSectionError}.jwtCookie`)
  }

  const exports = []

  exports.push(
    `const jwtCookie = ${JSON.stringify(resolveConfig.jwtCookie, null, 2)}`,
    ``,
    `export default jwtCookie`
  )

  return {
    code: exports.join('\r\n')
  }
}
