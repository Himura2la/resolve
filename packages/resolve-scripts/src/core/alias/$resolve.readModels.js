import { injectEnv } from 'json-env-extract'

import { message } from '../constants'
import resolveFile from '../resolve_file'
import importBabel from '../import_babel'

export default ({ resolveConfig, isClient }) => {
  if (!resolveConfig.readModels) {
    throw new Error(`${message.configNotContainSectionError}.readModels`)
  }

  const imports = [
    `import interopRequireDefault from "@babel/runtime/helpers/interopRequireDefault"`,
    ``
  ]
  const constants = [``]
  const exports = [``, `const readModels = []`, ``]

  for (let index = 0; index < resolveConfig.readModels.length; index++) {
    const readModel = resolveConfig.readModels[index]

    const name = readModel.name

    const projection = resolveFile(readModel.projection)

    const resolvers = resolveFile(readModel.resolvers)

    const storage = readModel.storage

    constants.push(`const name_${index} = ${JSON.stringify(name)}`)

    if (isClient) {
      const clientResolvers = Object.keys(importBabel(resolvers))

      constants.push(
        `const resolvers_${index} = {`,
        clientResolvers
          .map(commandName => `  ${commandName}() {}`)
          .join(',\r\n'),
        `}`
      )
    } else {
      imports.push(
        `import projection_${index} from ${JSON.stringify(projection)}`,
        `import resolvers_${index} from ${JSON.stringify(resolvers)}`,
        ``
      )
    }

    if (!isClient && readModel.storage) {
      constants.push(
        `const storage_${index} = ${injectEnv(storage)}`,
        `const storageAdapter_${index} = interopRequireDefault(`,
        `  require(storage_${index}.adapter)`,
        `).default`,
        `const storageOptions_${index} = storage_${index}.options`
      )
    }

    exports.push(`readModels.push({`, `  name: name_${index}`)
    if (!isClient) {
      exports.push(`, projection: projection_${index}`)
    }
    exports.push(`, resolvers: resolvers_${index}`)
    if (!isClient && readModel.storage) {
      exports.push(
        `, snapshot: {`,
        `    adapter: storageAdapter_${index},`,
        `    options: storageOptions_${index}`,
        `  }`
      )
    }
    exports.push(`})`, ``)
  }

  exports.push(`export default readModels`)

  return {
    code: [...imports, ...constants, ...exports].join('\r\n')
  }
}
