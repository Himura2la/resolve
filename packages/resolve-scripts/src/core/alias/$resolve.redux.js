import { message } from '../constants'

export default ({ resolveConfig }) => {
  if (!resolveConfig.redux) {
    throw new Error(`${message.configNotContainSectionError}.redux`)
  }

  const exports = []

  exports.push(
    `import store from ${JSON.stringify(resolveConfig.redux.store)}`,
    `import reducers from ${JSON.stringify(resolveConfig.redux.reducers)}`,
    `import middlewares from ${JSON.stringify(
      resolveConfig.redux.middlewares
    )}`,
    ``,
    `export default {`,
    `  store, reducers, middlewares`,
    `}`
  )

  return {
    code: exports.join('\r\n')
  }
}
