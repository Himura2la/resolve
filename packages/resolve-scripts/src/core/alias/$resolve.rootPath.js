import { message } from '../constants'

export default ({ resolveConfig }) => {
  if (resolveConfig.rootPath === undefined) {
    throw new Error(`${message.configNotContainSectionError}.rootPath`)
  }

  const exports = []

  exports.push(
    `const rootPath = ${JSON.stringify(resolveConfig.rootPath)}`,
    ``,
    `export default rootPath`
  )

  return {
    code: exports.join('\r\n')
  }
}
