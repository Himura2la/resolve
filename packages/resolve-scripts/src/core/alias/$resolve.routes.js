import { message } from '../constants'

export default ({ resolveConfig }) => {
  if (!resolveConfig.routes) {
    throw new Error(`${message.configNotContainSectionError}.routes`)
  }

  const exports = []

  exports.push(
    `import routes from ${JSON.stringify(resolveConfig.routes)}`,
    ``,
    `export default routes`
  )

  return {
    code: exports.join('\r\n')
  }
}
