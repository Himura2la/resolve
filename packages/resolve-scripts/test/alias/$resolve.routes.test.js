import path from 'path'

import alias from '../../src/core/alias/$resolve.routes'

describe('base config works correctly', () => {
  const resolveConfig = {
    routes: path.resolve(__dirname, 'files/testRoutes.js')
  }

  test('[client]', () => {
    expect(
      '\r\n' +
        alias({
          resolveConfig,
          isClient: true
        }).code +
        '\r\n'
    ).toMatchSnapshot()
  })

  test('[server]', () => {
    expect(
      '\r\n' +
        alias({
          resolveConfig,
          isClient: false
        }).code +
        '\r\n'
    ).toMatchSnapshot()
  })
})
