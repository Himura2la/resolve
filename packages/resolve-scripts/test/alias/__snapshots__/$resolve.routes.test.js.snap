// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base config works correctly [client] 1`] = `
"
import routes from \\"/home/mrcheater/resolve/packages/resolve-scripts/test/alias/files/testRoutes.js\\"

export default routes
"
`;

exports[`base config works correctly [server] 1`] = `
"
import routes from \\"/home/mrcheater/resolve/packages/resolve-scripts/test/alias/files/testRoutes.js\\"

export default routes
"
`;
